grommet/hpe-design-system
# Contain Layers to Example region

[Preview](https://deploy-preview-1283--keen-mayer-a86c8b.netlify.app/templates/filtering#filters-in-a-center-layer)



#### What does this PR do?
Allow Layers that are part of Examples to be contained to the dimensions of the example. This is done by creating refs within the internal Example component that can be consumed be the example as a prop. The ref is used as the target for the example Layer. This does require adding in a target to the example (which may not be desired in production code).  To mitigate confusion, I have placed a comment on every line where the ref or target is mentioned saying "[x] is for demo purposes only, remove in production". I have also cleaned up some console warnings re: prop types.

#### Where should the reviewer start?
aries-site/src/layouts/content/Example/Example.js

#### What testing has been done on this PR?
Tested with Filter examples and updated all Filter examples.

#### How should this be manually tested?
Check Filter page on deploy. The Layers should open in the bounds of the example region.

#### Any background context you want to provide?

#### What are the relevant issues?
Closes #1115 

#### Screenshots (if appropriate)

#### Should this PR be mentioned in Design System updates?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.
